### PR TITLE
chore: bump flutter/dart min sdk constraints

### PR DESCRIPTION
**Pull request by btrautmann**
## Description

Bumps minimum version constraint on Flutter/dart to 3.16.0/3.2.0 respectively. Doing this ahead of https://github.com/Betterment/alchemist/pull/115 should resolve the issues seen with the build there.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

**Comment by btrautmann** [comment]
Blocked by the issue mentioned at https://github.com/Betterment/alchemist/issues/104, looking into it...

File: test/src/golden_test_scenario_test.dart [hunk 1]
```diff
@@ -151,9 +151,9 @@ void main() {
         expect(
           mediaQuery,
           isA<MediaQueryData>().having(
-            (m) => m.textScaleFactor,
-            'textScaleFactor',
-            2.0,
+            (m) => m.textScaler,
+            'textScalar',
```
Comment (jeroen-meijer) [inline test/src/golden_test_scenario_test.dart:155]:
```suggestion
            'textScaler',
```

**Review by jeroen-meijer — DISMISSED** [review]
Minor comment, looks good otherwise.

Feel free to request another review whenever you're done adding commits, or merge as-is.

**Review by ClaireDavis — APPROVED** [review]
domain lgtm

**Review by Kirpal — APPROVED** [review]

**Pull request merged**